lib/mumukit/flow/adaptive_item.rb

```diff
@@ -0,0 +1,57 @@
+module Mumukit::Flow
+  module AdaptiveItem
+    include Mumukit::Flow::Node
+    include Mumukit::Flow::Suggesting
+
+    required :assignment
+    required :children
+    required :parent
+    required :tags
+
+    delegate :passed?, :submitter, to: :@assignment
+
+    def has?(tag)
+      tags.include?(tag)
+    end
+
+    private
+
+    def set_adaptive_assignment!(submitter)
+      @assignment = assignment_for submitter
+      @assignment.skip_if_pending!
+    end
+
+    def end_reached?
+      next_item.nil?
+    end
+
+    def next_item
+      next_items.first
+    end
+
+    def next_items
+      sorted_pending_siblings.select { |it| it.number > number }
+    end
+
+    def sorted_pending_siblings
+      pending_siblings.sort_by(&:number)
+    end
+
+    def pending_siblings
+      passed_items = passed_siblings_by(submitter).map { |assignment| assignment.item }
```

Comment:
I've replaced the block with (&:item).

Sadly we can't use `siblings` there because, while the actual `AdaptiveItem` does have an Assignment (due to `set_adaptive_assignment!`, which I commented earlier), the siblings don't - we have to query them explicitly. Remember an exercise doesn't have an assignment by itself in our domain model. Therefore, doing something like

```
def pending_siblings
   siblings.select(&:pending?)
end
```

doesn't work, because we can't know if they're pending without the assignments. That's why we need to use `passed_siblings_by`, which queries them.

This is less than ideal and I intend to optimize it somehow, but I haven't found a good way yet.